recording-daemon/decoder.c

```diff
@@ -115,7 +115,7 @@ static int decoder_got_frame(decoder_t *dec, AVFrame *frame, void *sp, void *dp)
 	if (metafile->mix_out) {
 		dbg("adding packet from stream #%lu to mix output", stream->id);
 		if (G_UNLIKELY(deco->mixer_idx == (unsigned int) -1))
-			deco->mixer_idx = mix_get_index(metafile->mix, ssrc);
+			deco->mixer_idx = mix_get_index(metafile->mix, ssrc, ssrc->stream->name);
```

Comment:
Have you considered using `ssrc->stream->id` instead of parsing out the `name` string? I don't know if that would actually work, but at least it looks like it might be a better option.